pkg/operator/status.go

```diff
@@ -289,9 +291,105 @@ func (optr *Operator) syncUpgradeableStatus() error {
 		coStatus.Message = "One or more machine config pools are updating, please see `oc get mcp` for further details"
 	}
 
+	// don't overwrite status if updating or degraded
+	if !updating && !degraded {
+		if supported, v := optr.isKubeletSkewSupported(); !supported {
+			coStatus.Reason = "KubeletSkewUnsupported"
+			coStatus.Message = fmt.Sprintf("one or more nodes have an unsupported kubelet version skew. Please see `oc get nodes` for details and upgrade all nodes so that they have a kubelet Version of at least %v.", getMinimalSkewSupportNodeVersion(v))
```

Comment:
I'm just noticing that [the enhancement][1] doesn't currently have much detail here.  There's some follow-up work still in flight [here][2] with:

> The Machine Config Operator should report a message in its reporting that admins should perform an action.

But based on my fuzzy recollection of some in-person meeting at some point, the MCO was going to warn folks who had outdated kubelets that they were behind, with hints about how to update.  Or maybe they were going to warn folks who were up against the boundary (so with the Kube-API-server's `Upgradeable=False` going off), with hints about how to update.  Staying silent until the nodes are _past_ the boundary seems like it's waiting too long.

[1]: https://github.com/openshift/enhancements/blame/3716ed25593447d9e624cb4fbabd90fcec66c58f/enhancements/update/eus-upgrades-mvp.md
[2]: https://github.com/openshift/enhancements/pull/800/files#diff-91c1b36fc770bc12ef9147ff77360aa00fb1f79d3df986a471d75d1c4a33891cR292-R293

Reply:
This goes hand in hand with kube-apiserver, but might be worth having 2 cases:

1) KubeletSkewUnsupported for people past the boundary (which I think can happen since this won't be backported to infinity?) 
2) KubeletSkewPresent (or some better name than what I can think of right now) where we are informationally saying: that we recommend upgrading your pool (maybe they are just 1 skew), let them know then and maybe they will just upgrade and save us all the trouble? 

Basically if there is skew you always say something, but you'll probably want to say different things

Reply:
@kikisdeliveryservice Follow your suggestion, what if change to the following check and warn the users there's going to be an unsupported version skew?
```go
minorVSkew, kasV, kubeletV := optr.isKubeletSkewSupported()
if minorVSkew > 2 {
	coStatus.Reason = "KubeletSkewUnsupported"
	coStatus.Message = fmt.Sprintf("one or more nodes have an unsupported kubelet version skew. Please see `oc get nodes` for details and upgrade all nodes so that they have a kubelet Version of at least %v.", getMinimalSkewSupportNodeVersion(v))
} else if minorVSkew == 2 {
	coStatus.Reason = "KubeletSkewPresent"
	coStatus.Message = fmt.Sprintf("current kubelet version %v will not be supported by newer kube-apiserver. Please upgrade the the kubelet first if plan to upgrade the kube-apiserver")
}
```

Reply:
Skew is also present if the diff is 1 and I think Trevor is pointing out that we should say something whenever there is a skew and not wait.

Reply:
There will be a skew regardless on upgrades. Is the intent to report to the user on an upgrade there is a skew?

Reply:
The enhancement is for >=2 skew... If we want to create a further enhancement for 4.10 for skew < 2 we can do that. It's out of scope of this PR.